python-sdk/src/astro/databases/snowflake.py

```diff
@@ -460,6 +468,21 @@ def is_native_autodetect_schema_available(  # skipcq: PYL-R0201
         )
         return is_file_type_supported and is_file_location_supported
 
+    def create_table(self, table: BaseTable, *args, **kwargs):
+        """Override create_table to add metadata columns to the table if specified in load_options"""
+        super().create_table(table, *args, **kwargs)
+        if self.load_options is None or not self.load_options.metadata_columns:
+            return
+        table_name = self.get_table_qualified_name(table)
+        metadata_columns = ",".join(
+            [
+                "IF NOT EXISTS " + f"{col} {self.METADATA_COLUMNS_DATATYPE[col]}"
+                for col in self.load_options.metadata_columns
+            ]
+        )
+        sql_statement = f"ALTER TABLE {table_name} ADD COLUMN {metadata_columns};"
+        self.hook.run(sql_statement, autocommit=True)
```

Comment:
@pankajkoti Just curious Why are we altering a table? Do you think we can do the required changes at creation time only? might save us one extra query execution.

Reply:
Have we tested this for the following scenarios? The table is created using pandas and the same table is used to load metadata. Would it work for this?

Reply:
Yes, tested this. This alter statement is after the create table, so this is applied to both whether the table is created by pandas or via infer schema and hence we are able to load metadata as those columns are created.